project/Build.scala

```diff
@@ -1384,14 +1384,36 @@ object Build {
 
   def testHtmlSettings[T](testHtmlKey: TaskKey[T], targetStage: Stage) = Seq(
       // We need to patch the system properties.
-      scalaJSJavaSystemProperties in Test in testHtmlKey ~= { base =>
-        val unsupported =
-          Seq("nodejs", "nodejs.jsdom", "phantomjs", "source-maps")
-        val supported =
-          Seq("typedarray", "browser")
-
-        base -- unsupported.map("scalajs." + _) ++
-            supported.map("scalajs." + _ -> "true")
+      scalaJSJSFiles in (Test, testHtmlKey) := {
+        val previousFiles = (scalaJSJSFiles in (Test, testHtmlKey)).value
+
+        val patchedSystemProperties = {
+          val base = (scalaJSJavaSystemProperties in Test).value
+          val unsupported =
+            Seq("nodejs", "nodejs.jsdom", "phantomjs", "source-maps")
+          val supported =
+            Seq("typedarray", "browser")
+
+          base -- unsupported.map("scalajs." + _) ++
+              supported.map("scalajs." + _ -> "true")
+        }
+
+        val code = s"""
+          var __ScalaJSEnv = {
+            javaSystemProperties: ${jsonToString(patchedSystemProperties.toJSON)}
+          };
+        """
+
+        val patchedSystemPropertiesFile =
+          new MemVirtualJSFile("setJavaSystemProperties.js").withContent(code)
+
+        // Replace the normal `setJavaSystemProperties.js` file with the patch
+        for (file <- previousFiles) yield {
+          if (file.path == "setJavaSystemProperties.js")
+            patchedSystemPropertiesFile
+          else
+            file
+        }
```

Comment:
Yeah, I just meant change it and read it again when you patch `scalaJSJSFiles`. Simply because it's more sbt-like.